packages/main/src/Link.ts

```diff
@@ -260,6 +298,20 @@ class Link extends UI5Element implements ITabbable {
 			&& this._dummyAnchor.protocol === loc.protocol);
 	}
 
+	_ensureNoIconOnlyLink() {
+		if (this._isIconOnly) {
+			throw new Error("Icon only links are not supported.");
+		}
+	}
+
+	get _isIconOnly() {
+		return (this.icon || this.endIcon) && !this.textContent?.trim();
+	}
+
+	get effectiveIconMode() {
+		return IconMode.Decorative;
```

Comment:
What is the purpose of this getter? Is it intended to be overridden by descendants? I can't think of any components that extend `ui5-link` right now. Just set Decorative manually in the template?